site/src/pages/WorkspaceSettingsPage/WorkspaceSettingsForm.tsx

```diff
@@ -40,21 +41,30 @@ export const WorkspaceSettingsForm: FC<{
     error,
   );
 
+  // We can't use `form.touched` unfortunately, because it only gets updated
+  // when a field loses focus, not when it gets changed.
+  const [nameModified, setNameModified] = useState(false);
```

Comment:
I haven't worked with Formik before, so I'm not sure what edge cases it has, but I would expect a derived value to work. Something like:
```ts
const nameModified = form.initialValues.name !== form.values.name
```
Are there any edge cases to that?

Reply:
huh! I didn't expect that to work given how `touched` works, but it totally does!